R/pset_to_gdrDF.R

```diff
@@ -0,0 +1,196 @@
+    assertthat::assert_that(is.integer(workers),
+                            msg = "workers parameter must be an integer. Default = 2L")
+
+    # FUNCTION TO MANIPULATE ENV IDENTIFIERS
+    setEnvForPSet()
+
+    # GET DOSE AND VIABILITY DATA & MELT INTO LARGE TABLE
+    dose_response <- .extractDoseResponse(pset = pharmacoset)
+
+    # ADD IN DURATION AND REFERENCE DIVISION TIME
+    dose_response_duration_refdivtime <- .createPseudoData(dose_response)
+
+    # REMOVE NEGATIVE VIABILITIES
+    dt <- .removeNegatives(dose_response_duration_refdivtime)
+
+    as.data.frame(dt)
+}
+
+#' Adjust environment variables to meet gDR standards
+#' 
+#' @export
+setEnvForPSet <- function() {
+    ## -- Set environment identifiers to map from our columns to gDR columns
+    gDRutils::reset_env_identifiers()
+
+    pgx_to_gdr_ids <- list(
+        cellline = "Clid",
+        cellline_name = "Clid",
+        drug = "DrugName",
+        drug_name = "DrugName",
+        concentration = "Concentration",
+        duration = "Duration",
+        barcode = "Barcode"
+    )
+    
+    invisible(Map(function(x, y) gDRutils::set_env_identifier(x, y), names(pgx_to_gdr_ids), pgx_to_gdr_ids))
+}
+
+#' Get PharmacoSet 
+#' @keywords internal
+.getPSet <- function(pset_name,
+                     psetDir = getwd(),
+                     canonical = FALSE,
+                     timeout = 600) {
+   
+    assertthat::assert_that(is.character(pset_name),
+                            msg = "pset_name parameter must be a character vector.")
+  
+    checkmate::assert_character(getwd())
+    checkmate::assert_flag(canonical)
+    checkmate::assert_numeric(timeout)
+
+    availPSets <- PharmacoGx::availablePSets(canonical = canonical)
+
+    pset_name_param <- if (pset_name %in% availPSets$"Dataset Name") {
+      availPSets[availPSets$"Dataset Name" == pset_name, "PSet Name"]
+      } else if (pset_name %in% availPSets$"PSet Name") {
+        pset_name    
+    } else {
+        stop(pset_name, 
+            " does not exist in the available PSets. Try one of the following:\n", 
+            paste(availPSets$`PSet Name`, collapse = '\n'))
+    }
+
+    # Check if PSet exists in directories where PSets are stored. 
+    # Read in if exists, download otherwise
+    pset <- if (file.exists(file.path(psetDir, paste0(pset_name_param,".rds")))) {
+      message("PSet exists in user-provided directory, reading .rds file")
+      readRDS(file.path(psetDir, paste0(pset_name_param,".rds")))
+      } else {
+        message("PSet does not exist in user-provided directory, downloading from database.")
+        PharmacoGx::downloadPSet(pset_name_param, saveDir = psetDir, timeout = timeout)
+      }
+    # Update PSet to new structure if not updated already.
+    PharmacoGx::updateObject(pset)
+}
+
+#' Get dose and viability readouts and melt into large data table
+#' @keywords internal
+#' @importFrom data.table `:=` setDF
+.extractDoseResponse <- function(pset) {
```

Comment:
it would be great to stick to the 50 lines per function rule from BiocCheck

Reply:
Done